go/internal/feast/basefeatureview.go

```diff
@@ -0,0 +1,43 @@
+package feast
+
+import (
+	"errors"
+	"fmt"
+	"github.com/feast-dev/feast/go/protos/feast/core"
+)
+
+type BaseFeatureView struct {
+	name       string
+	features   []*Feature
+	projection *FeatureViewProjection
+}
+
+func NewBaseFeatureView(name string, featureProtos []*core.FeatureSpecV2) *BaseFeatureView {
+	base := &BaseFeatureView{name: name}
+	features := make([]*Feature, len(featureProtos))
+	for index, featureSpecV2 := range featureProtos {
+		features[index] = NewFeatureFromProto(featureSpecV2)
+	}
+	base.features = features
+	base.projection = NewFeatureViewProjectionFromDefinition(base)
+	return base
+}
+
+func (fv *BaseFeatureView) withProjection(projection *FeatureViewProjection) (*BaseFeatureView, error) {
+	if projection.name != fv.name {
+		return nil, errors.New(fmt.Sprintf("The projection for the %s FeatureView cannot be applied because it differs in name. "+
```

Comment:
I've made this comment before, but can we please sweep through the errors and ensure that they are lower case and without trailing punctuation?